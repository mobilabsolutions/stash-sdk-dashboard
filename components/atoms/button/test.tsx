/* eslint-env jest */

import { testRender } from '../../../test_utils'
import PrimaryButton from './primary'
import SecondaryButton from './secondary'
import WarnButton from './warn'

it('PrimaryButton Should Render', () =>
  testRender(PrimaryButton, { lable: 'Login' }))

it('Disabled PrimaryButton Should Render', () =>
  testRender(PrimaryButton, { lable: 'Disabled', disabled: true }))

it('Full size PrimaryButton Should Render', () =>
  testRender(PrimaryButton, { lable: 'Disabled', isFullSize: true }))

it('SecondaryButton Should Render', () =>
  testRender(SecondaryButton, { lable: 'Cancel' }))

it('Disabled PrimaryButton Should Render', () =>
  testRender(SecondaryButton, { lable: 'Cancel', disabled: true }))

it('WarnButton Should Render', () =>
  testRender(WarnButton, { lable: 'Cancel' }))

it('Disabled WarnButton Should Render', () =>
  testRender(WarnButton, { lable: 'Cancel', disabled: true }))
