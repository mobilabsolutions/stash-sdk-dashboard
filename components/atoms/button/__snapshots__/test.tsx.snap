// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Disabled PrimaryButton Should Render 1`] = `
<button
  className="primary__Button-sc-1lwi4g7-0 hmcZQn"
  disabled={true}
  onClick={null}
  type="submit"
>
  <span />
</button>
`;

exports[`Disabled PrimaryButton Should Render 2`] = `
<button
  className="secondary__Button-j7mjyk-0 iLEulF"
  disabled={true}
  onClick={null}
  type="button"
>
  <span />
</button>
`;

exports[`Disabled WarnButton Should Render 1`] = `
<button
  className="warn__Button-sc-5cg2rp-0 gxotsG"
  disabled={true}
  onClick={null}
  type="submit"
>
  <span />
</button>
`;

exports[`Full size PrimaryButton Should Render 1`] = `
<button
  className="primary__Button-sc-1lwi4g7-0 fPlfTR"
  disabled={false}
  onClick={null}
  type="submit"
>
  <span />
</button>
`;

exports[`PrimaryButton Should Render 1`] = `
<button
  className="primary__Button-sc-1lwi4g7-0 hmcZQn"
  disabled={false}
  onClick={null}
  type="submit"
>
  <span />
</button>
`;

exports[`SecondaryButton Should Render 1`] = `
<button
  className="secondary__Button-j7mjyk-0 gxdnVF"
  disabled={false}
  onClick={null}
  type="button"
>
  <span />
</button>
`;

exports[`WarnButton Should Render 1`] = `
<button
  className="warn__Button-sc-5cg2rp-0 gxotsG"
  disabled={false}
  onClick={null}
  type="submit"
>
  <span />
</button>
`;
